src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Controller;
+
+use FOS\RestBundle\View\View;
+use Sylius\Bundle\ResourceBundle\Controller\RequestConfiguration;
+use Sylius\Bundle\ResourceBundle\Controller\ResourceController;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Resource\ResourceActions;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class ProductController extends ResourceController
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function showAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::SHOW);
+        /** @var ProductInterface $resource */
+        $resource = $this->findOr404($configuration);
+
+        $this->eventDispatcher->dispatch(ResourceActions::SHOW, $configuration, $resource);
+
+        $view = View::create($resource);
+
+        if ($configuration->isHtmlRequest()) {
+            $view = $this->prepareHtmlRequestView($resource, $configuration, $view);
+        }
+
+        return $this->viewHandler->handle($configuration, $view);
+    }
+
+    /**
+     * @param ProductInterface $resource
+     * @param RequestConfiguration $configuration
+     * @param View $view
+     *
+     * @return View
+     */
+    protected function prepareHtmlRequestView(ProductInterface $resource, RequestConfiguration $configuration, View $view)
+    {
+        $templateData = [
+            'configuration' => $configuration,
+            'metadata' => $this->metadata,
+            'resource' => $resource,
+            $this->metadata->getName() => $resource,
+        ];
+
+        if (
+            !$resource->isSimple() &&
```

Comment:
Creating `$product->isConfigurable()` method which is a shortcut for `!$product->isSimple()` would improve readability a lot.